File: Others/easy/RunLengthEncoding.py
def runLengthEncoding(string):
    curr_mult = 1

    new_str = []
    for i in range(len(string) - 1):
        if string[i] != string[i + 1] or curr_mult == 9:
            new_str.append(str(curr_mult) + string[i])
            curr_mult = 1
        else:
            curr_mult += 1

    return ''.join(new_str + [str(curr_mult) + string[-1]])

File: Golang/easy/run-length-encoding.go
package main

import "fmt"

func RunLengthEncoding(str string) string {
	newStr := ""
	currLength := 1
	currChar := str[0]

	for idx := 1; idx < len(str); idx += 1 {
		if str[idx] == currChar {
			currLength += 1
		} else {
			groups := currLength / 9
			lastGroup := currLength % 9
			for gr := 0; gr < groups; gr += 1 {
				newStr = newStr + string('9') + string(currChar)
			}
			if lastGroup > 0 {
				newStr = newStr + fmt.Sprint(lastGroup) + string(currChar)
			}
			currLength = 1
			currChar = str[idx]
		}
	}

	groups := currLength / 9
	lastGroup := currLength % 9
	for gr := 0; gr < groups; gr += 1 {
		newStr = newStr + string('9') + string(currChar)
	}
	if lastGroup > 0 {
		newStr = newStr + fmt.Sprint(lastGroup) + string(currChar)
	}

	return newStr
}
